moonraker/plugins/power.py

```diff
@@ -97,6 +104,14 @@ async def _handle_power_request(self, web_request):
 
     async def _process_request(self, device, req):
         if req in ["on", "off"]:
+            printing = await self._check_klippy_printing()
+            logging.info("Device locked: %s" % device.get_locked_while_printing())
+            logging.info("Klippy printing: %s" % printing)
+            if device.get_locked_while_printing() and printing:
+                self.server.error(f"Unable to change power for {device} " +
+                    "while printing")
+                dev_info = device.get_device_info()
+                return dev_info['status']
```

Comment:
I think you need to raise the exception.  When doing so it won't be necessary to return device status, the client will receive the error instead.

Reply:
Thanks for the feedback. I just changed it to raise an error